x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1195,3 +1195,30 @@ function deepMergeVars(original: any, override: any): any {
 
   return result;
 }
+
+export async function incrementPackageName(
+  soClient: SavedObjectsClientContract,
+  packageName: string
+) {
+  // Fetch all packagePolicies having the package name
+  const systemPackagePolicyData = await packagePolicyService.list(soClient, {
+    perPage: 1,
+    kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name: "${packageName}"`,
+  });
+
+  // Retrieve highest number appended to package policy name and increment it by one
+  const pkgPoliciesNamePattern = new RegExp(`${packageName}-(\\d+)`);
+
+  const pkgPoliciesWithMatchingNames = systemPackagePolicyData?.items
+    ? systemPackagePolicyData.items
+        .filter((ds) => Boolean(ds.name.match(pkgPoliciesNamePattern)))
+        .map((ds) => parseInt(ds.name.match(pkgPoliciesNamePattern)![1], 10))
+        .sort((a, b) => a - b)
```

Comment:
nit: I don't think the sort implementation is necessary?
```suggestion
        .sort()
```